### PR TITLE
Fixed deprecation warnings and index increment

### PR DESCRIPTION
**Pull request by vfn**
The if inside the generate methods would test index and then increment it, allowing it to go out of bounds.

### CODE REVIEW

**Comment by andreacremaschi** [comment]
Hey @vfn thank you for the PR
Oh, I see, the index thing was definitely wrong, wonder why it didn't pop out before.
I'll check why Travis is complaining before merging.

**Comment by vfn** [comment]
@andreacremaschi I forgot to update the travis file to use `xcode7.3`. The PR has been updated, build and tests are fine, but Travis is now crashing https://travis-ci.org/andreacremaschi/GEOSwift/builds/120500155

It looks like a xctool issue https://github.com/facebook/xctool/issues/666

File: GEOSwift/GEOS.swift [hunk 1]
```diff
@@ -184,9 +184,10 @@ public struct CoordinatesCollection: SequenceType {
     
     public func generate() -> AnyGenerator<Coordinate> {
         var index: UInt32 = 0
-        return anyGenerator {
+        return AnyGenerator {
+            index += 1
```
Comment (andreacremaschi) [inline GEOSwift/GEOS.swift]: This is wrong: `index` should be incremented AFTER accessing the element, not BEFORE, something like this:

```swift
    public func generate() -> AnyGenerator<T> {
        var index: Int32 = 0
        return AnyGenerator {
            if index < self.count {
                let item = self[index]
                index = index + 1
                return item
            }
            return nil
        }
    }
```

**Comment by vfn** [comment]
https://github.com/facebook/xctool/issues/684

**Comment by andreacremaschi** [comment]
thanks! :beers:

**Comment by vfn** [comment]
👍

**Pull request merged**